src/main/java/co/rsk/bitcoinj/script/RedeemScriptParserFactory.java

```diff
@@ -0,0 +1,120 @@
+import java.util.List;
+
+public class RedeemScriptParserFactory {
+
+    public static RedeemScriptParser get(List<ScriptChunk> chunks) {
+        if (chunks.size() < 4) {
+            // A multisig redeem script must have at least 4 chunks (OP_M PUB1 OP_N CHECK_MULTISIG)
+            return new NoRedeemScriptParser();
+        }
+
+        ParseResult result = extractRedeemScriptFromChunks(chunks);
+
+        if (result == null) {
+            return new NoRedeemScriptParser();
+        }
+
+        if (internalIsFastBridgeMultiSig(result.internalScript)) {
+            return new FastBridgeRedeemScriptParser(
+                result.scriptType,
+                result.internalScript,
+                chunks
+            );
+        } else if (internalIsStandardMultiSig(result.internalScript)) {
+            return new StandardRedeemScriptParser(
+                result.scriptType,
+                result.internalScript,
+                chunks
+            );
+        }
+        return new NoRedeemScriptParser();
+    }
+
+    private static ParseResult extractRedeemScriptFromChunks(List<ScriptChunk> chunks) {
+        ScriptChunk lastChunk = chunks.get(chunks.size() - 1);
+        if (isRedeemLikeScript(chunks)) {
+            return new ParseResult(chunks, ScriptType.REDEEM_SCRIPT);
+        }
+        if (lastChunk.data != null && lastChunk.data.length > 0) {
+            int lastByte = lastChunk.data[lastChunk.data.length - 1] & 0xff;
+            if (lastByte == OP_CHECKMULTISIG || lastByte == OP_CHECKMULTISIGVERIFY) {
+                ScriptParserResult result = ScriptParser.parseScriptProgram(lastChunk.data);
+                if (result.getException().isPresent()) {
+                    throw new ScriptException("Error trying to parse inner script", result.getException().get());
+                }
+                return new ParseResult(result.getChunks(), ScriptType.P2SH);
+            }
+        }
+        return null;
+    }
+
+    private static boolean isRedeemLikeScript(List<ScriptChunk> chunks) {
+        if (chunks.size() < 4) return false;
+        ScriptChunk chunk = chunks.get(chunks.size() - 1);
+        // Must end in OP_CHECKMULTISIG[VERIFY]
+        if (!chunk.isOpCode()) return false;
+        return chunk.equalsOpCode(OP_CHECKMULTISIG) || chunk.equalsOpCode(OP_CHECKMULTISIGVERIFY);
+    }
+
+    private static boolean hasRedeemScriptFormat(List<ScriptChunk> chunks) {
+        try {
+            // Second to last chunk must be an OP_N opcode and there should be
+            // that many data chunks (keys).
+            ScriptChunk m = chunks.get(chunks.size() - 2);
+            if (!m.isOpCode()) return false;
+            int numKeys = Script.decodeFromOpN(m.opcode);
+            if (numKeys < 1 || chunks.size() != 3 + numKeys) return false;
+            for (int i = 1; i < chunks.size() - 2; i++) {
+                if (chunks.get(i).isOpCode()) return false;
+            }
```

Comment:
Not sure if this is the expected behaviour for the `n` value or if it should be using `m`

Reply:
This function evaluates the format of the redeemScript, it should be `OP_N [PUBKEY] OP_N VERIFY` where the amount of `PUBKEY` should be the same as the value of the second `OP_N` (the n value).
To do this it verifies that:
- N is an OP_N
- there are N chunks for the pubkeys
- M is an OP_N